.github/workflows/pr_check_workflow.yml

```diff
@@ -1,25 +1,107 @@
 # This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
 # For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
 
-name: Node.js CI
+name: Build and test
 
 on:
   push:
     branches: [ main ]
   pull_request:
     branches: [ main ]
```

Comment:
Shouldn't we have this run on all branches? That would catch any potential unexpected merge issues/conflicts.